src/core/IO/Path.pm

```diff
@@ -277,13 +277,13 @@ my class IO::Path is Cool does IO {
             if nqp::iseq_s($part, $up) {
                 next unless $res-list;
                 nqp::pop_s($res-list);
-                $resolved = $res-list ?? $sep ~ nqp::join($sep, $res-list)
+                $resolved = $res-list ?? $!SPEC.catdir($res-list.flat)
                                       !! $empty;
```

Comment:
Are you sure that `.flat` is needed here? Doesn't look like the list is getting any unflat stuff added into it anywhere

Reply:
I think i need to pass something that matches the signature of catdir. The nqp::list_s() is not. I've copied this code from somewhere else where a List is needed. 

it should work now?